### PR TITLE
Skip machine-id if already present

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

The `systemd` package now includes a dummy `/etc/machine-id` file. Previously the installer tools were responsible for creating this file. Update the tools to only create the file if it is missing.

###### Change Log  

- Make `addMachineID()` skip if `/etc/machine-id` is already present.

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local  build of tools
- core-efi build in progress

### CODE REVIEW

**Review by anphel31 — APPROVED** [review]

File: toolkit/tools/imagegen/installutils/installutils.go [hunk 1]
```diff
@@ -758,9 +763,16 @@ func addMachineID(installChroot *safechroot.Chroot) (err error) {
 
 	ReportAction("Configuring machine id")
 
-	err = installChroot.UnsafeRun(func() error {
-		return file.Create(machineIDFile, machineIDFilePerms)
-	})
+	exists, err := file.PathExists(filepath.Join(installChroot.RootDir(), machineIDFile))
+	if err != nil {
+		err = fmt.Errorf("unable to check if machine-id exists. Error:\n%w", err)
```
Comment (neha170) [inline toolkit/tools/imagegen/installutils/installutils.go:768]: nit:
to keep it consistent with error handling in utlity, could you try
`err = fmt.Errorf("failed to check if machine-id exists:\n%w", err)`

Comment (dmcilvaney) [reply toolkit/tools/imagegen/installutils/installutils.go:768]: I think we use `Error:\n%w` about 50/50 in other code, and this file is all "Error: ". I'm fine with simplifying but we should do that in one pass I think.

Comment (neha170) [reply toolkit/tools/imagegen/installutils/installutils.go:768]: Yes, cleaning up the error logging in toolkit so it's uniform

Comment (dmcilvaney) [reply toolkit/tools/imagegen/installutils/installutils.go:768]:
```suggestion
		err = fmt.Errorf("failed to check if machine-id exists:\n%w", err)
```

**Pull request merged**